webapi/src/main/scala/org/knora/webapi/responders/admin/ProjectsResponderADM.scala

```diff
@@ -778,11 +778,12 @@ class ProjectsResponderADM(responderData: ResponderData) extends Responder(respo
             _ = if (shortcodeExists) {
                 throw DuplicateValueException(s"Project with the shortcode: '${createProjectRequest.shortcode}' already exists")
             }
+
             newProjectIRI: IRI = createProjectRequest.projectIri match {
-                case Some(customProjectIri) => customProjectIri.toString
-                case None => stringFormatter.makeRandomProjectIri(validatedShortcode)
+              case Some(customProjectIri) => stringFormatter.toSmartIriWithErr(customProjectIri, throw BadRequestException(s"Invalid project IRI")).toString
```

Comment:
`StringFormatter.toSmartIriWithErr` checks that it's a valid IRI, but not that it's a valid project IRI. It looks like there's a method for that: `StringFormatter.validateProjectIri`, which is already used a lot in `ProjectsRouteADM`. In general, Knora does input format validation in routes rather than in responders.

Reply:
@benjamingeer thanks for reviewing this. I just added the validator as you suggested for the custom project IRI. Since the same validation was missing in the case of custom list IRI, I also added that in this PR. I hope it is OK. Is there anything else you believe must be added?